Kickstarter-iOS/Views/PledgeCTAContainerView.swift

```diff
@@ -98,31 +89,81 @@ final class PledgeCTAContainerView: UIView {
       |> \.font .~ UIFont.ksr_caption1().bolded
       |> \.textColor .~ UIColor.ksr_dark_grey_500
       |> \.numberOfLines .~ 0
+
+    _ = self.activityIndicator
+      |> \.color .~ UIColor.ksr_dark_grey_500
+      |> \.hidesWhenStopped .~ true
   }
 
   // MARK: - View model
 
   override func bindViewModel() {
     super.bindViewModel()
 
-    self.vm.outputs.buttonTitleTextColor
+    self.viewModel.outputs.buttonTitleTextColor
       .observeForUI()
       .observeValues { [weak self] textColor in
         self?.pledgeCTAButton.setTitleColor(textColor, for: .normal)
       }
 
-    self.titleAndSubtitleStackView.rac.hidden = self.vm.outputs.stackViewIsHidden
-    self.titleLabel.rac.text = self.vm.outputs.titleText
-    self.subtitleLabel.rac.text = self.vm.outputs.subtitleText
-    self.pledgeCTAButton.rac.backgroundColor = self.vm.outputs.buttonBackgroundColor
-    self.pledgeCTAButton.rac.title = self.vm.outputs.buttonTitleText
-    self.spacer.rac.hidden = self.vm.outputs.spacerIsHidden
+    self.viewModel.outputs.rootStackViewAnimateIsHidden
+      .observeValues { [weak self] isHidden in
+        let duration = isHidden ? 0.0 : 0.18
+        let alpha: CGFloat = isHidden ? 0.0 : 1.0
+        UIView.animate(withDuration: duration, animations: {
+          _ = self?.rootStackView
+            ?|> \.alpha .~ alpha
+        })
+      }
+
+    self.viewModel.outputs.activityIndicatorIsAnimating
+      .observeValues { isAnimating in
+        let announcementMessage = isAnimating ? "Loading project state" : "Loading project state complete"
+
+        UIAccessibility.post(notification: .announcement, argument: announcementMessage)
```

Comment:
@jamielynnroth @dannyalright need some guidance here. What this code does is it posts a notification for users with Voice Over enabled, such that if the activity indicator is loading, it will say "Loading project state". If the activity indicator is not loading (ie. the request has completed) it will say "Loading project state complete".

When testing this, I noticed that the requests are coming back so quickly that only the "Loading project state complete" announcement is actually heard. I'm wondering if this is useful at all then? I imagine it maybe being useful for Voice Over users under bad network conditions, but if it's a very small edge case it might not be worth having this at all. Thoughts?

Reply:
Hmm. Outside of this immediate question: I think we should have guidelines for when and when _not_ to have VoiceOver.

Does the loading VoiceOver string impede or block the fully-loaded VoiceOver string from being said? If not, I'd be in favour for keeping it how you have it (both _loading_ and _loading complete_).

Reply:
My understanding is that we have a pretty low number of users with voice-over enabled so I wouldn't want to over-engineer this for a tiny minority, but something else we can do is add a delay to this signal of like a second or so when we know that voice over is enabled?

Reply:
I think if we want to implement this announcement I'll add a new signal for it, but @dannyalright I'm not sure I understand your question. The loading Voice Over string cannot be heard most of the time because the fetch happens so quickly - so all a user hears is "Loading Complete". This feels a bit out of context to me and not really that useful unless you actually have slow signal and hear the previous announcement of "Loading project state", so I'm inclined to potentially remove this announcement all-together.

Reply:
Fair enough. Let's get rid of it!